packages/dds/tree/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -727,44 +957,343 @@ export class ModularChangeFamily
+				metadata,
+				constraintState,
+			);
+
+			if (rebasedNode !== undefined) {
+				setInChangeAtomIdMap(rebasedNodes, newId, rebasedNode);
+			}
+		}
+
+		return rebasedFields;
+	}
+
+	// This processes fields which have no base changes but have been invalidated by another field.
+	private rebaseFieldsWithoutBaseChanges(
+		rebasedFields: FieldChangeMap,
+		rebasedNodes: ChangeAtomIdMap<NodeChangeset>,
+		crossFieldTable: RebaseTable,
+		genId: IdAllocator,
+		metadata: RebaseRevisionMetadata,
+	): void {
+		for (const [revision, localId, fieldKey] of crossFieldTable.affectedNewFields.keys()) {
+			const nodeId: NodeId | undefined =
+				localId !== undefined ? { revision, localId } : undefined;
+
+			const fieldMap = fieldMapFromNodeId(rebasedFields, rebasedNodes, nodeId);
+
+			const fieldChange = fieldMap.get(fieldKey);
+			assert(fieldChange !== undefined, "Cross field key registered for empty field");
+
+			if (crossFieldTable.rebasedFields.has(fieldChange)) {
+				// This field has already been processed because there were base changes.
+				continue;
+			}
+
+			// This field has no changes in the base changeset, otherwise it would have been added to `crossFieldTable.rebasedFields`
+			// when processing fields with both base and new changes.
+			const rebaseChild = (
+				child: NodeId | undefined,
+				baseChild: NodeId | undefined,
+				stateChange: NodeAttachState | undefined,
+			): NodeId | undefined => {
+				assert(baseChild === undefined, "There should be no base changes in this field");
+				return child;
+			};
+
+			const handler = getChangeHandler(this.fieldKinds, fieldChange.fieldKind);
+			const baseFieldChange: FieldChange = {
+				...fieldChange,
+				change: brand(handler.createEmpty()),
+			};
+
+			const fieldId: FieldId = { nodeId, field: fieldKey };
+			const rebasedField = handler.rebaser.rebase(
+				fieldChange.change,
+				baseFieldChange.change,
+				rebaseChild,
+				genId,
+				new RebaseManager(crossFieldTable, baseFieldChange, fieldId),
+				metadata,
+			);
+
+			const rebasedFieldChange: FieldChange = { ...fieldChange, change: brand(rebasedField) };
+			fieldMap.set(fieldKey, rebasedFieldChange);
+
+			// TODO: Deduplicate these lines with other rebase locations.
+			crossFieldTable.baseFieldToContext.set(baseFieldChange, {
+				newChange: fieldChange,
+				baseChange: baseFieldChange,
+				rebasedChange: rebasedFieldChange,
+				fieldId,
+				baseNodeIds: [],
+			});
+			crossFieldTable.rebasedFields.add(rebasedFieldChange);
+		}
+	}
+
+	// This processes fields which have no new changes but have been invalidated by another field.
+	private rebaseFieldsWithoutNewChanges(
+		rebasedFields: FieldChangeMap,
+		rebasedNodes: ChangeAtomIdMap<NodeChangeset>,
+		crossFieldTable: RebaseTable,
+		genId: IdAllocator,
+		metadata: RebaseRevisionMetadata,
+	): void {
+		const baseChange = crossFieldTable.baseChange;
+		for (const [revision, localId, fieldKey] of crossFieldTable.affectedBaseFields.keys()) {
```

Comment:
I'm confused: why are we pulling from `affectedBaseFields` here? Shouldn't we be pulling from `affectedNewFields` instead?